specutils/io/default_loaders/wcs_fits.py

```diff
@@ -79,6 +94,20 @@ def wcs1d_fits_loader(file_obj, spectral_axis_unit=None, flux_unit=None,
 
     if spectral_axis_unit is not None:
         wcs.wcs.cunit[0] = str(spectral_axis_unit)
+    elif wcs.wcs.cunit[0] == '' and 'WAT1_001' in header:
+        # Try to extract from IRAF-style card or use Angstrom as default.
+        wat_dict = dict((rec.split('=') for rec in header['WAT1_001'].split()))
+        unit = wat_dict.get('units', 'Angstrom')
+        if hasattr(u, unit):
```

Comment:
> This _might_ be better as a `try`/`except` with `u.Unit(unit)` only because the value error is more descriptive and might be better to let the user know possible conversions that weren't taken?

The original purpose was simply to replace the previous code, which was really just checking for
```
if wat_dict['units'] == 'angstroms':
    unit = u.angstrom
```
with something slightly more generic. Accepting `centimeters` or `microns` is rather a side product, but `angstroms` is so ubiquitous (in fact the only case considered for the IRAF headers) that it would generate too much noise as a warning.
But you are probably right that the final `wcs.wcs.cunit[0]` should still be validated before the `wcs` would fail further down.